src/entity/Unit.ts

```diff
@@ -1285,37 +1283,72 @@ export function inRange(unit: IUnit, target: Vec2): boolean {
   return math.distance(unit, target) <= unit.attackRange;
 }
 
-// return boolean signifies if unit should abort their turn
-export async function runTurnStartEvents(unit: IUnit, prediction: boolean = false, underworld: Underworld): Promise<boolean> {
-  // Note: This must be a for loop instead of a for..of loop
-  // so that if one of the onTurnStartEvents modifies the
-  // unit's onTurnStartEvents array (for example, after death)
-  // this loop will take that into account.
-  let abortTurn = false;
-  for (let i = 0; i < unit.onTurnStartEvents.length; i++) {
-    const eventName = unit.onTurnStartEvents[i];
-    if (eventName) {
+export async function startTurnForUnits(units: IUnit[], underworld: Underworld, prediction: boolean) {
+  // Trigger start turn events
+  for (let unit of units) {
+    await runTurnStartEvents(unit, underworld, prediction);
```

Comment:
Should these run in parallel? Like the following example code 
```
const turnStartPromises = [];
for(let unit of units){
turnStartPromises.push(runTurnStartEvents(...etc etc))
}
await race(timeout, Promise.all(turnStartPromises));
```
This way if theres an on turn start event that does something that takes time like moving a unit they won't trigger one at a time.  And we should use raceTimeout here to protect against bugs where the promise never returns.

Reply:
Yes, you're right, will implement 👍